src/GroupManager.php

```diff
@@ -44,13 +69,41 @@ class GroupManager {
 
   /**
    * Constructs an GroupManager object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The service providing information about bundles.
    */
-  public function __construct(ConfigFactoryInterface $config_factory) {
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeBundleInfoInterface $entity_type_bundle_info) {
     $this->configFactory = $config_factory;
+    $this->entityTypeBundleInfo = $entity_type_bundle_info;
     $this->refreshGroupMap();
   }
 
   /**
+   * Magic getter.
+   *
+   * @param string $property
+   *   The property being gotten.
+   *
+   * @return mixed
+   *   The property value.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when an invalid property is passed.
+   */
+  public function __get($property) {
```

Comment:
It's a little trick to do lazy loading of the relation map. The magic getter only fires if the property doesn't exist. So it will fire, the relation map will be calculated and the property will be set. The next time the property is accessed it will exist and the magic getter won't be fired again.